README.md

```diff
@@ -1,14 +1,14 @@
 # enchannel
 
-Enchannel is a light spec for setting up communications between a frontend, like the notebook or jupyter-sidecar, and a backend kernel (the runtime, like Python, Julia, or R).
+Enchannel is a light spec for setting up communications between a frontend, like the [notebook](https://github.com/jupyter/notebook) or [jupyter-sidecar](https://github.com/nteract/jupyter-sidecar), and a backend kernel (the runtime, like Python, Julia, or R).
 
 [![enchannel](https://cloud.githubusercontent.com/assets/836375/12282043/b19bb16e-b960-11e5-8661-ce2111ec0417.png)](https://cloud.githubusercontent.com/assets/836375/12282043/b19bb16e-b960-11e5-8661-ce2111ec0417.png)
 
-The core functionality of the notebook is to send messages from a frontend to a backend, and a backend to a frontend ([or many frontends](https://github.com/nteract/jupyter-sidecar)).
+The core functionality of the notebook is to send messages from a frontend to a backend, and from a backend to a frontend ([or many frontends](https://github.com/nteract/jupyter-sidecar)).
 
 In the case of the Jupyter/IPython notebook, it communicates over websockets (which in turn reach out to 0MQ).
 
-What if you want to serve the same HTML and Javascript for the notebook application itself while being able to work in a native 0MQ environment? What if websockets are fairly restricted in your *ahem corporate* environment and you need to do e.g. `POST`ing + Server Sent Events?
+What if you want to serve the same HTML and Javascript for the notebook application itself while being able to work in a native 0MQ environment? What if websockets are fairly restricted in your *ahem corporate* environment and you need to send data via `POST` or stream updates using server-sent events?
```

Comment:
Ok, so this last one is a bit strange (on purpose). The directionality is like this:

```
        ---- Server Sent Events --->
Kernel                               Frontend
        <--- HTTP POST data --------
```

Mostly because they don't have the nicety of websockets. The other alternative is polling for data and posting new data (yes, this is an awkward flow for Jupyter).